packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -1105,6 +1109,7 @@ const getLegendProps = (data: Data[], layout: Partial<Layout> | undefined) => {
 
   return {
     legends,
-    hideLegend: layout?.showlegend === false ? true : hideLegends,
+    hideLegend:
+      layout?.showlegend === false || (layout?.showlegend !== true && legends.length < 2) ? true : hideLegends,
```

Comment:
>!== [](http://example.com/codeflow?start=58&length=3)

=== ?

Reply:
`!==` is used here to handle the case where `layout.showlegend` is undefined and there are less than 2 traces.